tenacity/__init__.py

```diff
@@ -257,19 +257,27 @@ def copy(
         after=_unset,
         before_sleep=_unset,
         reraise=_unset,
+        retry_error_cls=_unset,
+        retry_error_callback=_unset,
     ):
         """Copy this object with some parameters changed if needed."""
-        if before_sleep is _unset:
-            before_sleep = self.before_sleep
+
+        def first_set(first, second):
+            return second if first is _unset else first
```

Comment:
There should be no need to redefine this on each call, you can just move it as a static method.

Reply:
Thank you for the review.

My thinking was that since the method is very limited in scope, making it a local function would be clearer to the reader than defining it at the module level. And `copy()` being called probably not that often, the performance overhead seemed acceptable.

Making a static method is too cumbersome and defeats the purpose of the function IMHO:
```python
            reraise=BaseRetrying._first_set(reraise, self.reraise),
            retry_error_cls=BaseRetrying._first_set(
                retry_error_cls, self.retry_error_cls
            ),
```

In the end I moved it as a module-level function, if you're OK with that.

Reply:
Having it as a module-level function disallow inheritance to override the method.
I don't think it's a big deal though, so that's fine.